sync15-adapter/src/bso_record.rs

```diff
@@ -0,0 +1,287 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde::de::DeserializeOwned;
+use serde::ser::Serialize;
+use serde_json;
+use error;
+use base64;
+use std::ops::{Deref, DerefMut};
+use std::convert::From;
+use key_bundle::KeyBundle;
+use util::ServerTimestamp;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BsoRecord<T> {
+    pub id: String,
+
+    // It's not clear to me if this actually can be empty in practice.
+    // firefox-ios seems to think it can...
```

Comment:
It's not one of the publicly-documented fields of a BSO at [1], so absolutely it can be empty.

[1] https://mozilla-services.readthedocs.io/en/latest/storage/apis-1.5.html#basic-storage-object